birdvoxpaint/util.py

```diff
@@ -0,0 +1,225 @@
+import datetime
+import numpy as np
+
+import librosa
+from librosa.util.exceptions import ParameterError
+
+import joblib
+
+
+sample2frame = lambda x, window, hop: int((x - window) / hop) + 1
+frame2sample = lambda x, window, hop: int((x - 1) * hop) + window
+
+def block_stream(filename=None, y=None, sr=None,
+                 frame_length=2048, hop_length=512,
+                 segment_duration=1, n_blocks=1, full_frames=True):
```

Comment:
i don't think `n_blocks` is the right abstraction here. i would prefer an upper bound in the memory footprint that an upper bound on the number of iterations

Reply:
`n_blocks` was really just referring to `n_jobs` which is how you were calculating it before (1 block per job). The nomenclature is confusing too though so there's definitely room for improvement there. Maybe `n_seg_per_block` is right? idkkkkk

Reply:
yes, it's `n_seg_per_block`. But i would prefer to call it `n_jobs` if that what it's going to relate to. Another option is to exclude it from the kwarg list. With high probability, we aren't going to need it.